core/src/main/java/nu/yona/server/analysis/entities/DayActivity.java

```diff
@@ -0,0 +1,66 @@
+package nu.yona.server.analysis.entities;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import nu.yona.server.entities.RepositoryProvider;
+import nu.yona.server.goals.entities.Goal;
+
+@Entity
+@Table(name = "DAY_ACTIVITIES")
+public class DayActivity extends IntervalActivity
+{
+	public static DayActivityRepository getRepository()
+	{
+		return (DayActivityRepository) RepositoryProvider.getRepository(DayActivity.class, UUID.class);
+	}
+
+	@OneToMany(cascade = CascadeType.ALL)
+	private List<Activity> activities;
+
+	// Default constructor is required for JPA
+	public DayActivity()
+	{
+		super(null, null, null, null);
+	}
+
+	public DayActivity(UUID id, UUID userAnonymizedID, UUID goalID, ZonedDateTime zonedStartOfDay, List<Activity> activities)
```

Comment:
This one can be private, right? It's only used from the ``createInstance`` method.